README.md

```diff
@@ -2,6 +2,86 @@
 
 The job runner is started by the batch/resource manager system and is the component that actually executes the SDK module.  It also provides the callback handler and pushes logs to the execution engine logging system.
 
+## Callback Server Mode
+
+It is possible to run the callback server in a standalone mode.  This can be used to speed up
+testing of SDK apps (avoiding a full make test cycle) or to allow some automated operations.
+The callback server can be launched in a few ways.
+
+### Native Launch
+
+To launch the callback server natively you can do the following.
+
+```
+pip install .
+# Set the DOCKER_HOST if this doesn't work out of the box
+export DOCKER_HOST=unix://$HOME/.docker/run/docker.sock
+
+# Set job dir where work output will go
+export JOB_DIR=/full/path/to/work/area
+
+# Set tokens and URL
+export KB_AUTH_TOKEN="xxxxxxxxx"
+export KB_ADMIN_AUTH_TOKEN="xxxxxxxxxxxx"
+export KB_BASE_URL=https://ci.kbase.us/services
+
+# Optional
+export KB_REF_DATA=/path/to/local/refdata
+
+# Launch the callback server
+python -m JobRunner.Callback
+```
+
+### Container Launch
+
+The callback server can also be launched via a container.  This may be useful
+where the local python and what works with the JobRunner are incompatible.
+Note that the JOB_DIR needs to be accessible by docker.  Also you must pass through
+the docker socket to the container.  The path to the socket can vary depending on
+the container runtime and how the container runtime is configured.
+
+```
+export JOB_DIR=/full/path/to/work/area
+export KB_AUTH_TOKEN="xxxxxxxxx"
+export KB_ADMIN_AUTH_TOKEN="xxxxxxxxxxxx"
+
+docker run --name cb -d \
+   -e KB_AUTH_TOKEN \
+   -e KB_ADMIN_AUTH_TOKEN \
+   -e KB_BASE_URL=https://ci.kbase.us/services \
+   -e CALLBACK_PORT=9999 \
+   -e JOB_DIR \
+   -v $JOB_DIR:$JOB_DIR \
+   -v /var/run/docker.sock:/run/docker.sock \
+   -p 9999:9999 \
+   ghcr.io/kbase/jobrunner:latest-rc
+
+export SDK_CALLBACK_URL=http://localhost:9999
+```
+
+## Development and Testing the Job Runner
+
+Here is a quick start guide for running test for the Job Runner code.
+See the SDK guide for information about running test of SDK apps.
+
+```
+pip install -r requirements.txt -r requirements-dev.txt
+
+# Set the DOCKER_HOST if this doesn't work out of the box
+export DOCKER_HOST=unix://$HOME/.docker/run/docker.sock
+
+# Set tokens and URL
+export KB_AUTH_TOKEN="xxxxxxxxx"
+export KB_ADMIN_AUTH_TOKEN="xxxxxxxxxxxx"
+export KB_BASE_URL=https://ci.kbase.us/services
+
+# Set ref data to an area acceessible by Docker
+export KB_REF_DATA=/path/to/local/refdata
+
+make mock
+make testimage
+make test
```

Comment:
RE Make test

* Nose is not installed
* coverage is not installed
* Won't run on python 3.11 (                         ^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'collections' has no attribute 'Callable')

Reply:
If you install requirements-dev.txt it will add those.  I will make a comment that this is currently working against Python 3.9